tests/scripts/thread-cert/Cert_9_2_19_PendingDatasetGet.py

```diff
@@ -0,0 +1,300 @@
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import unittest
+
+import mesh_cop
+import thread_cert
+from pktverify.consts import  MGMT_PENDING_GET_URI, MGMT_PENDING_SET_URI, NM_CHANNEL_TLV, NM_COMMISSIONER_ID_TLV, NM_COMMISSIONER_SESSION_ID_TLV, NM_STEERING_DATA_TLV, NM_BORDER_AGENT_LOCATOR_TLV, NM_PAN_ID_TLV, NM_NETWORK_NAME_TLV, NM_NETWORK_MESH_LOCAL_PREFIX_TLV, NM_PSKC_TLV, NM_ACTIVE_TIMESTAMP_TLV, NM_CHANNEL_MASK_TLV, NM_EXTENDED_PAN_ID_TLV, NM_NETWORK_MASTER_KEY_TLV, NM_SECURITY_POLICY_TLV, NM_PENDING_TIMESTAMP_TLV, NM_DELAY_TIMER_TLV
+from pktverify.packet_verifier import PacketVerifier
+from pktverify.null_field import nullField
+
+COMMISSIONER = 1
+LEADER = 2
+
+# Test Purpose and Description:
+# -----------------------------
+# The purpose of this test case is to verify Leader's and active Commissioner's behavior via
+# MGMT_PENDING_GET request and response
+#
+# Test Topology:
+# -------------
+# Commissioner
+#    |
+#  Leader
+#
+# DUT Types:
+# ----------
+#  Leader
+#  Commissioner
+
+
+class Cert_9_2_19_PendingDatasetGet(thread_cert.TestCase):
+    SUPPORT_NCP = False
+
+    TOPOLOGY = {
+        COMMISSIONER: {
+            'name': 'COMMISSIONER',
+            'mode': 'rdn',
+            'panid': 0xface,
+            'router_selection_jitter': 1,
+            'allowlist': [LEADER]
+        },
+        LEADER: {
+            'name': 'LEADER',
+            'mode': 'rdn',
+            'panid': 0xface,
+            'router_selection_jitter': 1,
+            'allowlist': [COMMISSIONER]
+        },
+    }
+
+    def test(self):
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[COMMISSIONER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[COMMISSIONER].get_state(), 'router')
+        self.simulator.get_messages_sent_by(LEADER)
+
+        self.collect_leader_aloc(LEADER)
```

Comment:
why do we need to collect leader ALOC? I believe leader ALOC is fixed in test.

Reply:
added const